docs/core/deploying/native-aot.md

```diff
@@ -0,0 +1,84 @@
+
+## Publish Native AOT - CLI
+
+Publish a native AOT application using the [dotnet publish](../tools/dotnet-publish.md) command.
+
+01. Add `<PublishAot>true</PublishAot>` to your project file.
+
+    This will produce a native AOT app and shows PublishAot compatibility warnings during build.
+
+    ```xml
+    <PropertyGroup>
+        <PublishAot>true</PublishAot>
+    </PropertyGroup>
+    ```
+
+02. Publish the app for for a specific runtime identifier using `dotnet publish -r <RID>`. There are pre-requisites that are needed to be installed for publishing a native AOT application. Some examples are given below for each platform.
+
+    For Windows pre-requisites, install [Visual Studio 2022](https://visualstudio.microsoft.com/vs/community/), including Desktop development with C++ workload. The following example publishes the app for Windows as a native AOT application on a machine with the required pre-requisites installed.
+
+    `dotnet publish -r win-x64 -c Release`
+
+    For Linux pre-requisites, install clang and developer packages for libraries that .NET runtime depends on.
+
+    *Ubuntu (16.04+)*
+
+    ```sh
+    sudo apt-get install clang zlib1g-dev
+    ```
+
+    *Alpine (3.14+)*
+
+    ```sh
+    sudo apk add clang gcc lld musl-dev build-base zlib-dev
+    ```
+
+    The following example publishes the app for Linux as a native AOT application. A native AOT binary produced on Linux machine is only going to work on same or newer Linux version than the build machine Linux version. For example, native AOT binary produced on Ubuntu 20.04 is going to run on Ubuntu 22.04, but it is not going to run on Ubuntu 16.04.
+
+    `dotnet publish -r linux-arm64 -c Release`
+
+The app will be available in the publish directory and will contain all the code needed to run in it, including the coreclr runtime.
+
+## Impact of using the Native AOT deployment
+
+Native AOT applications are [self-contained](index.md#publish-self-contained) and don't use any shared libraries, as is the case with the default [framework-dependent deployment model](index.md#publish-framework-dependent). Consider the native AOT deployment model when the benefits to deployment and the ability to optimize a single app outweigh the benefits of sharing for certain types of workloads. The gain is most significant for workloads with a high number of deployed instances: cloud infrastructure, hyper-scale services, popular apps, or games.
+
+Native AOT applications comes with a few fundamental limitations and compatibility issues. The key limitations include:
+
+- No dynamic loading (for example, `Assembly.LoadFile`)
+- No runtime code generation (for example, `System.Reflection.Emit`)
+- No C++/CLI
+- No built-in COM
+- Requires trimming, which has [limitations](trimming/incompatibilities.md)
+
+The first release of native AOT in .NET 7 has additional limitations. These include:
+
+- Should be targeted for console type applications.
```

Comment:
> console type applications

We should be more specific. ASP.NET Core apps are console apps.

Reply:
Yes, it is important to exclude the perception that ASP.NET Core apps are possible.

Is there a suitable phrasing to convey this? I was going with VS project templates that seems to restrict to non-ASP.NET apps.

Reply:
```suggestion
- Should be targeted for console type applications (not ASP.NET Core).
```